### PR TITLE
Kusto pools attached database configuration

### PR DESCRIPTION
**Pull request by ItaySofer**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [x] update existing version for new feature  [answer - still in private preview version]
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet. [answer - 7/2021]
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet. [answer - 7/2021]
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [x] No, no need to refresh for updates in this PR
[answer - No SDKs where published yet as we are in private preview, will be published for public preview]
### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s) [answer - We are still in private preview where these breaking changes are allowed]
  - Adding a new API version
  -  [ ] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by ItaySofer** [comment]
@ruowan please note we are in private preview version so breaking changes are allowed according to ARM guidlines here - HTTP://AKA.MS/AZBREAKINGCHANGESPOLICY.
Can the "Swagger BreakingChange" please be skipped?

**Comment by ruowan** [comment]
Your PR is in public repo, it's not private review. Has this api version 2021-04-04-preview already published?

**Comment by ItaySofer** [comment]
@ruowan It only starts rolling out now for private preview customers. Do you mean if the SDKs where published?

**Comment by idear1203** [comment]
In this case, I would suggest to hold on the PR until these APIs are in public preview and deployed worldwide

**Comment by ItaySofer** [comment]
@idear1203 we have a request by ARM to have an approved (not merged) Swagger PR before merging ARM manifest changes.
We need these changes for our development process.
Is it possible to approve this PR for us not to be unblocked on ARM manifest deployments, and only complete it before public preview?

Also, we will have some more Swagger changes for Swagger completeness needed for public preview.
So there might be some more PRs of this kind or we will add to it later on.

**Comment by idear1203** [comment]
Make sense. @ruowan could you please help with a review and add "DO NOT MERGE" label to this PR? Thanks in advance!

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-04-01-preview/kustoPool.json
Comment (markcowl) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-04-01-preview/kustoPool.json:1406]: null?  will this list never be pageable?

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-04-01-preview/kustoPool.json
Comment (markcowl) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-04-01-preview/kustoPool.json:1423]: parameters should appear in the order they appear in the uri:
- by path segments, in order
- query parameters
- body parameters

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-04-01-preview/kustoPool.json
Comment (markcowl) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-04-01-preview/kustoPool.json:1469]: same comment on parameter ordering

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-04-01-preview/kustoPool.json
Comment (markcowl) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-04-01-preview/kustoPool.json:1523]: please provide x-ms-long-running-operation-options: http://azure.github.io/autorest/extensions/#x-ms-long-running-operation-options

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-04-01-preview/kustoPool.json
Comment (markcowl) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-04-01-preview/kustoPool.json:1537]: parameter order

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-04-01-preview/kustoPool.json
Comment (markcowl) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-04-01-preview/kustoPool.json:1573]: add options, please

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-04-01-preview/kustoPool.json
Comment (markcowl) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-04-01-preview/kustoPool.json:1589]: parameter ordering

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-04-01-preview/kustoPool.json
Comment (markcowl) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-04-01-preview/kustoPool.json:1620]: not pageable?

Why is this modeled as a POST and not a GET?

Comment (markcowl) [reply specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-04-01-preview/kustoPool.json:1620]: Is this a long-running operation?

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-04-01-preview/kustoPool.json
Comment (markcowl) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-04-01-preview/kustoPool.json:1638]: parameter ordering

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-04-01-preview/kustoPool.json
Comment (markcowl) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-04-01-preview/kustoPool.json:1636]: Add x-ms-long-running-operation-options, please

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-04-01-preview/kustoPool.json
Comment (markcowl) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-04-01-preview/kustoPool.json:2648]: why does this resource have a location?

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-04-01-preview/kustoPool.json
Comment (markcowl) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-04-01-preview/kustoPool.json:2691]: What is the form of the entries in this list?  Are they resource ids, simple names?  A sample would be useful in the description.

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-04-01-preview/kustoPool.json
Comment (markcowl) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-04-01-preview/kustoPool.json:2712]:
```suggestion
      "description": "The attached database configuration properties.",
```

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-04-01-preview/kustoPool.json
Comment (markcowl) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-04-01-preview/kustoPool.json:2726]: an example of what this data should look like would be helpful

**Review by markcowl — CHANGES_REQUESTED** [review]
Also, you need to resolve the api-version issue - generally you cannot update an api-version after it has been deployed - there should eb a new api-version.

**Comment by ItaySofer** [comment]
> Also, you need to resolve the api-version issue - generally you cannot update an api-version after it has been deployed - there should eb a new api-version.

@markcowl we are in private preview version so breaking changes are allowed according to ARM guidelines here - HTTP://AKA.MS/AZBREAKINGCHANGESPOLICY.
Also, it is the first private preview version, not a private preview version for breaking changes of a product that is already in GA.
Can the changes stay in this API version?

We are doing the Swagger changes incrementally because we have a request by ARM to have an approved (not merged) Swagger PR before merging ARM manifest changes.
We need these changes for our development process and can't open a new API version with all the new capabilities at once, because that will drastically decrease our development speed.
All Swagger changes will be complete before public preview for "Swagger Completeness".

**Comment by ItaySofer** [comment]
@markcowl Can you please advise regarding the new api version issue described in my previous comment?

**Comment by ItaySofer** [comment]
> > Also, you need to resolve the api-version issue - generally you cannot update an api-version after it has been deployed - there should eb a new api-version.
> 
> @markcowl we are in private preview version so breaking changes are allowed according to ARM guidelines here - [HTTP://AKA.MS/AZBREAKINGCHANGESPOLICY](HTTP://AKA.MS/AZBREAKINGCHANGESPOLICY).
> Also, it is the first private preview version, not a private preview version for breaking changes of a product that is already in GA.
> Can the changes stay in this API version?
> 
> We are doing the Swagger changes incrementally because we have a request by ARM to have an approved (not merged) Swagger PR before merging ARM manifest changes.
> We need these changes for our development process and can't open a new API version with all the new capabilities at once, because that will drastically decrease our development speed.
> All Swagger changes will be complete before public preview for "Swagger Completeness".

@markcowl  Can you please advise regarding the new api version issue described the comment above?

**Pull request closed without merging**
